plugin/trino-openlineage/src/test/java/io/trino/plugin/openlineage/TestMarquezIntegration.java

```diff
@@ -0,0 +1,117 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+
+import static io.trino.plugin.openlineage.OpenLineageListenerQueryRunner.createOpenLineageRunner;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestMarquezIntegration
+        extends AbstractTestQueryFramework
+{
+    private static MarquezServer server;
+    private static String marquezURI;
+    private static final String trinoURI = "http://trino-integration-test:1337";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        server = closeAfterClass(new MarquezServer());
+        marquezURI = server.getMarquezUri().toString();
+
+        return createOpenLineageRunner(ImmutableMap.<String, String>builder()
+                        .put("openlineage-event-listener.transport.type", "HTTP")
+                        .put("openlineage-event-listener.transport.url", server.getMarquezUri().toString())
+                        .put("openlineage-event-listener.trino.uri", trinoURI)
+                        .buildOrThrow());
+    }
+
+    @Test
+    void testCreateTableAsSelectFrom()
+            throws Exception
+    {
+        String outputTable = "test_create_table_as_select_from_table";
+
+        @Language("SQL") String createTableQuery = format(
+                "CREATE TABLE %s AS SELECT * FROM tpch.tiny.nation",
+                outputTable);
+
+        assertQuerySucceeds(createTableQuery);
+
+        String queryId = computeActual(format(
+                "SELECT query_id FROM system.runtime.queries WHERE query LIKE '%s'",
+                createTableQuery))
+                .getMaterializedRows()
+                .getLast()
+                .getField(0)
+                .toString();
+
+        // Wait for marquez to process event
+        Thread.sleep(5000);
+
+        URI trino = new URI(trinoURI);
+
+        HttpClient client = HttpClient.newHttpClient();
```

Comment:
We could re-use the client across multiple tests right ?

Reply:
yes, moved it into private static final class attribute